src/librustc_lsan/lib.rs

```diff
@@ -17,4 +17,5 @@
             reason = "internal implementation detail of sanitizers",
             issue = "0")]
 
+#[allow(unused_extern_crates)]
```

Comment:
The `potentially_unused_imports` could be kept until normal late lints are run and then use the `tcx` to do those checks.

The crate id is available here: https://github.com/rust-lang/rust/blob/445077963c55297ef1e196a3525723090fe80b22/src/librustc_resolve/build_reduced_graph.rs#L260 and could be stored in the `ExternCrate` variant.